crates/sui-indexer-alt/src/handlers/obj_info_temp.rs

```diff
@@ -0,0 +1,694 @@
+        Ok(diesel::insert_into(obj_info_temp::table)
+            .values(stored)
+            .on_conflict_do_nothing()
+            .execute(conn)
+            .await?)
+    }
+
+    // TODO: Add tests for this function.
+    async fn prune(
+        &self,
+        from: u64,
+        to_exclusive: u64,
+        conn: &mut db::Connection<'_>,
+    ) -> Result<usize> {
+        use sui_indexer_alt_schema::schema::obj_info_temp::dsl;
+
+        let to_prune = self
+            .pruning_lookup_table
+            .get_prune_info(from, to_exclusive)?;
+
+        if to_prune.is_empty() {
+            self.pruning_lookup_table.gc_prune_info(from, to_exclusive);
+            return Ok(0);
+        }
+
+        // For each (object_id, cp_sequence_number), find and delete its immediate predecessor
+        let values = to_prune
+            .iter()
+            .map(|(object_id, seq_number)| {
+                let object_id_hex = hex::encode(object_id);
+                format!("('\\x{}'::BYTEA, {}::BIGINT)", object_id_hex, seq_number)
+            })
+            .collect::<Vec<_>>()
+            .join(",");
+
+        let query = format!(
+            "
+            WITH modifications(object_id, cp_sequence_number) AS (
+                VALUES {}
+            )
+            DELETE FROM obj_info_temp oi
+            USING modifications m
+            WHERE oi.{:?} = m.object_id
+              AND oi.{:?} = (
+                SELECT oi2.cp_sequence_number
+                FROM obj_info_temp oi2
+                WHERE oi2.{:?} = m.object_id
+                  AND oi2.{:?} < m.cp_sequence_number
+                ORDER BY oi2.cp_sequence_number DESC
+                LIMIT 1
+              )
+            ",
+            values,
+            dsl::object_id,
+            dsl::cp_sequence_number,
+            dsl::object_id,
+            dsl::cp_sequence_number,
+        );
+
+        let rows_deleted = sql_query(query).execute(conn).await?;
+        self.pruning_lookup_table.gc_prune_info(from, to_exclusive);
+        Ok(rows_deleted)
+    }
+}
+
+impl FieldCount for ProcessedObjInfo {
+    const FIELD_COUNT: usize = StoredObjInfoTemp::FIELD_COUNT;
+}
+
+impl TryInto<StoredObjInfoTemp> for &ProcessedObjInfo {
+    type Error = anyhow::Error;
+
+    fn try_into(self) -> Result<StoredObjInfoTemp> {
+        match &self.update {
+            ProcessedObjInfoUpdate::Insert(object) => {
+                StoredObjInfoTemp::from_object(object, self.cp_sequence_number as i64)
+            }
+            ProcessedObjInfoUpdate::Delete(object_id) => Ok(StoredObjInfoTemp {
+                object_id: object_id.to_vec(),
+                cp_sequence_number: self.cp_sequence_number as i64,
+                owner_kind: None,
+                owner_id: None,
+                package: None,
+                module: None,
+                name: None,
+                instantiation: None,
+            }),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
I copied the tests over as well to sanity check that I didn't change anything other than naming.